src/ffi/types.rs

```diff
@@ -3,10 +3,111 @@ use log::Level;
 use telio_crypto::KeyDecodeError;
 use telio_utils::map_enum;
 
-use std::ffi::c_void;
+use anyhow::anyhow;
+
+use std::{ffi::c_void, panic::RefUnwindSafe};
 
 use crate::device::{AdapterType, Error as DevError, Result as DevResult};
 
+pub trait TelioEventCb: Send + Sync + std::fmt::Debug {
+    fn event(&self, payload: String);
+}
+
+pub trait TelioLoggerCb: Send + Sync + std::fmt::Debug {
+    fn log(&self, log_level: TelioLogLevel, payload: String);
+}
+
+pub trait TelioProtectCb: Send + Sync + RefUnwindSafe + std::fmt::Debug {
+    fn protect(&self, payload: i32);
+}
+
+pub type FFIResult<T> = Result<T, TelioError>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum TelioError {
+    #[error("UnknownError: {0}")]
+    UnknownError(#[source] anyhow::Error),
+    #[error("InvalidKey")]
+    InvalidKey,
+    #[error("BadConfig")]
+    BadConfig,
+    #[error("LockError")]
+    LockError,
+    #[error("InvalidString")]
+    InvalidString,
+    #[error("AlreadyStarted")]
+    AlreadyStarted,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[allow(clippy::enum_variant_names)] // Otherwise clippy complains about all variants having "Tun" as their suffix
```

Comment:
Without this line the compilers/linters complain that all variants end with `Tun`